app/src/main/java/com/example/footprnt/Map/MapFragment.java

```diff
@@ -239,6 +244,11 @@ public boolean onTouch(View v, MotionEvent event) {
                 return false; // Pass on the touch to the map or shadow layer.
             }
         });
+        googleApiClient = new GoogleApiClient.Builder(getContext())
+                .addConnectionCallbacks(this)
```

Comment:
I like the use of implementing callbacks in the class you're using the variable. Abstraction is another pillar of object oriented programming and having a good understanding of it is super helpful